src/main/java/com/google/sps/data/AvailabilityTimeSlot.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package availability;
+
+/**
+ * An AvailabilityTimeSlot represents a table entry on the availability page that has a date and
+ * time (utcEncoding), a visual represenation of the time (timeString), and a class that is set
+ * based on whether or not the slot has already been selected in a previous visit to the page (green
+ * = selected = 'table-success').
+ */
+public class AvailabilityTimeSlot {
```

Comment:
Consider autovalue for this as well.

https://github.com/google/auto/tree/master/value

Here's what the current version of the code might look like with autovalue:

```java
  import com.google.auto.value.AutoValue;

  @AutoValue
  abstract static class AvailabilityTimeSlot {
    abstract String utcEncoding();
    abstract LocalTime time();
    abstract LocalDate date();
    abstract String cssClass();

    static AvailabilityTimeSlot create(String utcEncoding, LocalTime time, LocalDate date, String cssClass) {
      return new AutoValue_AvailabilityTimeSlot(utcEncoding, time, date, cssClass);
    }
  }
```

I can give the pod a primer on AutoValue if would be helpful. I find it incredibly useful both to simplify code, and also as a tool for thinking about problems, since I find the best code strongly distinguishes between simple value types (i.e. above), and classes which do lots of work using value types (i.e. a servlet class).

Reply:
Done